test/tests/functional/pbs_equiv_classes.py

```diff
@@ -1606,3 +1606,312 @@ class instead of the class it was in when it was queued.
+
+        # delete jid3 and check that jid1 goes back to R state
+        self.server.deljob(jid3, wait=True)
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid1)
+        self.scheduler.log_match("Number of job equivalence classes: 1",
+                                 max_attempts=10, starttime=self.t)
+
+    def test_multiple_node_multiple_job_order(self):
+        """Test jobs that come later in query order on multi-node
+        setup are added to its own equivalence class.This is
+        multinode test for test_premption2"""
+
+        # create 2 nodes
+        a = {'resources_available.ncpus': 3}
+        self.server.create_vnodes('vn', a, 2, self.mom, usenatvnode=False)
+
+        # create express queue
+        q = {'queue_type': 'e', 'started': 't',
+             'enabled': 't', 'priority': 150}
+        self.server.manager(MGR_CMD_CREATE, QUEUE, q, id='expressq')
+
+        # start 2 jobs of same equivalence class on node 1
+        j1 = Job(TEST_USER)
+        j1.set_attributes({'Resource_List.select': '1:ncpus=1:vnode=vn[0]'})
+        jid1 = self.server.submit(j1)
+
+        time.sleep(1)
+
+        j1 = Job(TEST_USER)
+        j1.set_attributes({'Resource_List.select': '1:ncpus=1:vnode=vn[0]'})
+        jid2 = self.server.submit(j1)
+
+        # start 2 jobs of same equivalence class on node 2
+        j2 = Job(TEST_USER)
+        j2.set_attributes(
+            {'Resource_List.select': '1:ncpus=1:vnode=vn[1]',
+             'Resource_List.walltime': 600})
+        jid3 = self.server.submit(j2)
+
+        time.sleep(1)
+
+        j2 = Job(TEST_USER)
+        j2.set_attributes(
+            {'Resource_List.select': '1:ncpus=1:vnode=vn[1]',
+             'Resource_List.walltime': 600})
+        jid4 = self.server.submit(j2)
+
+        time.sleep(1)
+
+        """Suspend 1 job from each node with a job of type of
+        different equivalence class"""
+
+        je = Job(TEST_USER)
+        je.set_attributes(
+            {'Resource_List.select': '1:ncpus=2:vnode=vn[0]',
+             'queue': 'expressq'})
+        jid5 = self.server.submit(je)
+
+        je = Job(TEST_USER)
+        je.set_attributes({'Resource_List.select': '1:ncpus=2:vnode=vn[1]',
+                           'queue': 'expressq'})
+        jid6 = self.server.submit(je)
+
+        # Check jid2 and jid4 are suspended
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid1)
+        self.server.expect(JOB, {'job_state': 'S'}, id=jid2)
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid3)
+        self.server.expect(JOB, {'job_state': 'S'}, id=jid4)
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid5)
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid6)
+
+        # check nuber of equivalence classes = 4
+        # 1 for j1d, 1 for jid2, 1 for jid3, 1 for jid4,
+        # and 1 for jid5 and jid6
+        self.scheduler.log_match("Number of job equivalence classes: 4",
```

Comment:
I think your comment is right, it says there should be 5 but you are checking for 4. are you getting 4 equiv classes?